dashboard/src/pages/macros.tsx

```diff
@@ -136,18 +121,25 @@ const Macros = () => {
                   row.name as string,
                   []
                 );
-                const newMacros = macros.map((macro) => {
-                  if (macro.name !== row.name) {
-                    return macro;
+
+                queryClient.setQueryData(
+                  ['instance', selectedInstance?.uuid, 'macroList'],
+                  (oldData: MacroEntry[] | undefined) => {
```

Comment:
We can probably simplify these using question marks and more idiomatic javascripy